rotkehlchen/inquirer.py

```diff
@@ -920,6 +924,112 @@ def find_curve_pool_price(
 
         return (total_assets_value / total_supply) * (10 ** lp_token.get_decimals())
 
+    def find_gearbox_price(
+            self,
+            token: EvmToken,
+    ) -> Price | None:
+        evm_manager = self.get_evm_manager(chain_id=token.chain_id)
+        node_inquirer = evm_manager.node_inquirer
+        globaldb = GlobalDBHandler()
+        underlying_token = None
+
+        farming_tokens = {token.farming_pool_token for token in read_gearbox_data_from_cache(token.chain_id)[0].values()}  # noqa: E501
+        with globaldb.conn.read_ctx() as cursor:
+            maybe_underlying_tokens = globaldb.fetch_underlying_tokens(
+                cursor=cursor,
+                parent_token_identifier=evm_address_to_identifier(
+                    address=token.evm_address,
+                    chain_id=token.chain_id,
+                    token_type=EvmTokenKind.ERC20,
+                ),
+            )
+
+        if token in farming_tokens:
+            farming_contract = EvmContract(
+                address=token.evm_address,
+                abi=node_inquirer.contracts.abi('GEARBOX_FARMING_POOL'),
+                deployed_block=0,  # not used here
+            )
+            try:
+                lp_token = farming_contract.call(node_inquirer=node_inquirer, method_name='stakingToken')  # noqa: E501
+                lp_token = deserialize_evm_address(lp_token)
+            except (RemoteError, BlockchainQueryError) as e:
+                log.error(f'Failed to query stakingToken method in Gearbox Pool. {e!s}')
+                return None
+        else:
+            lp_token = token.evm_address
+
+        lp_contract = EvmContract(
+            address=lp_token,
+            abi=node_inquirer.contracts.abi('GEARBOX_LP'),
+            deployed_block=0,  # not used here
+        )
+        if maybe_underlying_tokens is None or len(maybe_underlying_tokens) != 1:
+            # underlying token not recorded in the DB. Ask the chain
+            try:
+                remote_underlying_token = lp_contract.call(node_inquirer, 'underlyingToken')
+            except (RemoteError, BlockchainQueryError) as e:
+                log.error(f'Failed to query underlying token method in {node_inquirer.chain_name} Gearbox pool. {e!s}')  # noqa: E501
```

Comment:
this logic seems to be for the case where the asset doesn't have the underlying token right? Can you move it to its own function in the gearbox evm module? something like `ensure_gearbox_lp_underlying_tokens`. Maybe a better name. This will help clearing the logic for the reader